platforms/core-configuration/configuration-cache/src/integTest/groovy/org/gradle/configurationcache/isolated/IsolatedProjectsFixture.groovy

```diff
@@ -251,17 +272,45 @@ class IsolatedProjectsFixture {
             projectsConfigured(paths)
             runsTasks = false
             loadsOnStore = false
-            models.addAll(paths.collect { new ModelDetails(it, 1) })
+            models.addAll(paths.collect { new ModelRequestExpectation(it, 1) })
+        }
+
+        /**
+         * One model is created for each of the given projects. The projects will also be configured
+         */
+        void modelsCreated(Class<?> modelType, String... paths) {
+            projectsConfigured(paths)
+            runsTasks = false
+            loadsOnStore = false
+            models.addAll(paths.collect { new ModelRequestExpectation(it, [modelType.name]) })
         }
 
         /**
          * The given number of models are created for the given project. The project will also be configured
          */
         void modelsCreated(String path, int count) {
```

Comment:
Should we get rid of this method? I think there's something good about explicitly listing the model requests.

We can do it as a follow-up PR.